File: Homework 3/JAVA/Driver.java
import java.util.Scanner; 

public class Driver {

	private static Scanner input;

	public static void main(String[] args) {

		Boolean test = false;
		String s = "";
		String street, city, state;
		int zip = 0;
		input = new Scanner(System.in);

			System.out.println("Enter a street address (number and name): ");
			street = input.nextLine();
		
		while (test == false) {
			System.out.println("Enter a city: ");
			s = input.nextLine();
			// check for alphabetic
			if (CheckAlphaString.check(s) == true && s != "") {
				test = true;
			}
		}
			city = s;
			test = false;
			
		while (test == false) {
			System.out.println("Enter a state: ");
			s = input.nextLine();
			// check for alphabetic
			if (CheckAlphaString.check(s) == true && s != "") {
				test = true;
			}
		}
		state = s;
		
			System.out.println("Enter a zipcode: ");
			zip = input.nextInt();
		
		AlphaString cityAlpha = new AlphaString(city);
		SingleName cityName = new SingleName(cityAlpha);
		AlphaString stateAlpha = new AlphaString(state);
		SingleName stateName = new SingleName(stateAlpha);
		Address address = new Address(street, cityName, stateName, zip);
		
		System.out.printf("The full address is: %n" );
		System.out.printf( address.getStreet() + "%n" );
		System.out.printf( address.getCity() +  ", " +  address.getState() + " " + 
		address.getZip() + "%n");
		
	}
}

File: Homework 3/C++/Driver.cpp

#include <iostream>
#include "SingleName.h"
#include "CheckAlphaString.h"
#include "Address.h"

using namespace std;

int main()
{
	Address address;
	char input[100];
	string street;
	SingleName city;
	SingleName state;
	int zip;
	AlphaString alpha;
	SingleName single;
	CheckAlphaString check;

	cout << "Enter a street address (number and name): ";
	cin.ignore();
	cin.getline(input, 100);
	address.setStreet(input);


	cout << "Enter a city: ";
	do
	{
		cin >> input;
		alpha.set(input);
		if (check.check(alpha.get()) == false)
		{
			cout << "Please do not enter non alphabetic characters." << endl;
			cout << "Please try again: ";
		}
	} while (check.check(alpha.get()) == false);

	single.set(alpha);
	address.setCity(single);
	alpha.set("");

	cout << "Enter a state: ";
	do
	{
		cin >> input;
		alpha.set(input);
		if (check.check(alpha.get()) == false)
		{
			cout << "Please do not enter non alphabetic characters." << endl;
			cout << "Please try again: ";
		}
	} while (check.check(alpha.get()) == false);

	single.set(alpha);
	address.setState(single);
	alpha.set("");

	cout << "Enter a zip-code: ";
	cin >> zip;

	address.setZip(zip);


	cout << endl;
	cout << "The full address is: ";
	cout << endl;
	cout << address.getStreet();
	cout << endl;
	cout << address.getCity().get().get() << ", " << address.getState().get().get() << " " << address.getZip();
	cout << endl;

}

